From d19ff2f46d997ea6be96a19cd172e8a72b8b8f68 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 1 Dec 2020 17:27:21 +0100
Subject: [PATCH 1/1] riscv: qemu: enable distro boot from scsi

Booting via distro boot fails for:

    qemu-system-riscv64
    -drive if=none,file=sct-riscv64.img,format=raw,id=mydisk \
    -device ich9-ahci,id=ahci -device ide-hd,drive=mydisk,bus=ahci.0

Enable distro booting from an attached SCSI disk.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/configs/qemu-riscv.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/qemu-riscv.h b/include/configs/qemu-riscv.h
index aed9a4ae99..a2f33587c2 100644
--- a/include/configs/qemu-riscv.h
+++ b/include/configs/qemu-riscv.h
@@ -35,6 +35,7 @@
 #define BOOT_TARGET_DEVICES(func) \
 	func(QEMU, qemu, na) \
 	func(VIRTIO, virtio, 0) \
+	func(SCSI, scsi, 0) \
 	func(DHCP, dhcp, na)
 
 #include <config_distro_bootcmd.h>
-- 
2.29.2

